proto/v1/structure/languages.proto

```diff
@@ -0,0 +1,111 @@
+syntax = "proto3";
+
+package org.oppia.proto.v1.structure;
+
+import "proto/v1/structure/versions.proto";
+
+option java_package = "org.oppia.proto.v1.structure";
+option java_multiple_files = true;
+
+// A list of language code.
+enum LanguageCode {
+  LANGUAGE_CODE_UNSPECIFIED = 0;
+  ENGLISH = 1;
+  ARABIC = 2;
+  HINDI = 3;
+  HINGLISH = 4;
+  BRAZILIAN_PORTUGUESE = 5;
+}
+
+// NOTE: Update LanguageProtosVersion if the structure of this proto is modified.
+// Proto representing a block of (subtitled) HTML.
+message SubtitledHtml {
+  LanguageProtosVersion protos_version = 1;
+
+  // The content ID for this SubtitledHtml. This is unique within a State.
+  string content_id = 2;
+
+  // The HTML content of the SubtitledHtml.
+  string html = 3;
+}
+
+// NOTE: Update LanguageProtosVersion if the structure of this proto is modified.
+// Structure of the recorded voiceovers. 
+message RecordedVoiceovers {
+  LanguageProtosVersion protos_version = 1;
+
+  // A list of mapping from language codes to voiceover bundles (which map individual content IDs
+  // to their voiceovers).
+  repeated VoiceoverContentMapping voiceover_language_mapping = 2;
```

Comment:
converting this to `map<string, VoiceoverContentMapping >` from repeated helped me to get the [proto format in the correct structure](https://gist.github.com/anandwana001/e9c68fe2901ed2a9afec48c7cfa617e3#file-exploration-proto-L12). 

```diff
- repeated VoiceoverContentMapping voiceover_language_mapping = 2;
+ map<string, VoiceoverContentMapping> voiceover_language_mapping = 2;
```

With repeating values, we cannot hold the `content_id` and all we left with the language code and its respective voiceover, which is not the whole required data. 

<pre>
recorded_voiceovers:
      voiceovers_mapping:
        <b>ca_placeholder_3:</b> {}
        <b>content:</b>
          en:
            duration_secs: 0.0
            file_size_bytes: 57996
            filename: content-en-3iibmhu8je2.mp3
            needs_update: false
         <b>default_outcome: </b>
          en:
            duration_secs: 0.0
            file_size_bytes: 55733
            filename: feedback-en-s0w3v9gslr9.mp3
            needs_update: false
         <b>feedback_1: </b>
          en:
            duration_secs: 0.0
            file_size_bytes: 41549
            filename: feedback-en-kt33y1e2rwg.mp3
            needs_update: false
         <b>feedback_2: </b>
          en:
            duration_secs: 0.0
            file_size_bytes: 38524
            filename: feedback-en-vvixng5ajds.mp3
            needs_update: false
         <b>hint_1: </b>
          en:
            duration_secs: 0.0
            file_size_bytes: 47188
            filename: hint-en-jca2lxjibh.mp3
            needs_update: false
         <b>rule_input_4: </b> {}
         <b>rule_input_5: </b> {}
         <b>solution: </b> {}
</pre>

Reply:
What I go so far with representing as a map:
```
    recorded_voiceovers {
      voiceover_language_mapping {
        key: "ca_buttonText_0"
        value {
        }
      }
      voiceover_language_mapping {
        key: "content"
        value {
          voiceover_content_mapping {
            key: "en"
            value {
              filename: "test_audio_1_en.mp3"
              file_size_bytes: 95426
            }
          }
          voiceover_content_mapping {
            key: "hi-en"
            value {
              filename: "test_audio_1_hi_en.mp3"
              file_size_bytes: 132719
            }
          }
        }
      }
```
If we use `repeated VoiceoverContentMapping voiceover_language_mapping = 2;`, how does the above proto data looks like?

Reply:
`voiceover_language_mapping ` is removed from the structure but it is there in the new version. 

```
message RecordedVoiceovers {
  option (org.oppia.proto.v1.versions.structure_proto_version_type) = LANGUAGE_PROTOS_VERSION;

  // The version defining the internal structure of this message.
  org.oppia.proto.v1.versions.LanguageProtosVersion protos_version = 1;

  // A list of mappings from language to voiceover content mappings (see the message's documentation
  // for specifics on the latter).
  repeated VoiceoverContentMapping voiceover_content_mapping = 2;
}

// Represents a mapping of SubtitledText content IDs to voiceovers for a particular language.
message VoiceoverContentMapping {
  option (org.oppia.proto.v1.versions.structure_proto_version_type) = LANGUAGE_PROTOS_VERSION;

  // The language corresponding to this mapping.
  LanguageType language = 1;

  // The mapping of SubtitledText content IDs to their corresponding voiceover files.
  map<string, VoiceoverFile> voiceover_content_mapping = 2;
}

// Represents an audio file that, when played, reads out content text in a particular language.
message VoiceoverFile {
  option (org.oppia.proto.v1.versions.structure_proto_version_type) = LANGUAGE_PROTOS_VERSION;

  // The name of the voiceover audio file. This will never be a URL (and can always be expected to
  // just be a filename). The audio file is stored in GCS static file storage and must be downloaded
  // or streamed separately by the client. The client can expect that this file is immutable on the
  // server. Further, the client can also assume that all audio files are always of the format MP3.
  // There's no support for transcoding audio at a given bitrate, it's always fixed based on the
  // uploaded bitrate (as managed by Oppia's backend).
  string filename = 1;

  // The size of the voiceover file, in bytes.
  uint32 file_size_bytes = 2;

  // The duration of the voiceover audio track, in seconds.
  float duration_secs = 3;
}
```